src/compile/data/optimizers.ts

```diff
@@ -73,7 +74,10 @@ export class MoveParseUp extends BottomUpOptimizer {
         parent.merge(node);
       } else {
         // don't swap with nodes that produce something that the parse node depends on (e.g. lookup)
-        if (fieldIntersection(parent.producedFields(), node.dependentFields())) {
+        if (
+          parent instanceof PivotTransformNode ||
+          fieldIntersection(parent.producedFields(), node.dependentFields())
```

Comment:
How about we add a check to `fieldIntersection` to not allow `PivotTransformNode` as a parent. We might even be able to enforce this with the typing system. 

I think we are missing a check in `run` below, don't we?

Reply:
It looks like `fieldIntersection` only accepts two `Set<string>` arguments, so I'm not sure the best way of doing that.

For `run` below, the only other ones I saw that called `hasIntersection` or `fieldIntersection` checked checked whether the node was a certain type (`TimeUnitNode` in `RemoveDuplicateTimeUnits` and `BinNode` in `MergeBins`). Let me know if I'm missing a check somewhere though

Reply:
Ahh, good point. I guess we could change the function to take two nodes so that we can ensure our assertion. I think that would be a good idea. We can still keep the existing function since we use it in our tests. You can just wrap it with another function. 

In `MergeBins.run` we only check the child, but not the parent. Is that sufficient?

Reply:
Updated with that function `nodeFieldIntersection`, and I also added it to `MergeBins.run` because I wasn't sure if it would be necessary to check